presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/LocalProperties.java

```diff
@@ -87,6 +88,11 @@ else if (!(property instanceof ConstantProperty)) {
         return builder.build();
     }
 
+    public static <T> boolean matchesAll(List<LocalProperty<T>> actuals, List<LocalProperty<T>> desired)
+    {
+        return match(actuals, desired).equals(nCopies(desired.size(), Optional.empty()));
```

Comment:
I don't think the count is relevant. All you want is to make sure that all of them are `Optional.empty()`.